core/src/main/java/org/apache/calcite/rel/rules/AggregateJoinTransposeRule.java

```diff
@@ -146,12 +146,22 @@ private static boolean isAggregateSupported(Aggregate aggregate, boolean allowFu
     return true;
   }
 
+  // OUTER joins are supported for group by without aggregate functions
+  private boolean isJoinSupported(final Join join, final Aggregate aggregate) {
+    return join.getJoinType() != JoinRelType.FULL
```

Comment:
`&&` has higher precedence than `||`, hence this will not avoid execution of this rule for FULL OUTER joins in all cases. This should be rewritten to:
```
return join.getJoinType() == JoinRelType.INNER ||
   aggregate.getAggCallList().isEmpty() && join.getJoinType() != JoinRelType.FULL;
```
Please update comment above to describe why full outer join is not supported.

Reply:
> has higher precedence than ||
Parenthesis were left out by mistake. Thanks for noticing this. I'll update the pull request.